File: 0118-pascals-triangle/0118-pascals-triangle.cpp
class Solution {
public:
    vector<vector<int>> generate(int numRows) {
        vector<vector<int>> res;

        for (int i = 0; i < numRows; ++i){
            vector<int> row(i + 1, 1);  // 建立長度 i+1，預設為 1 的列

            // 填中間的數值
            for (int j = 1; j < i; ++j){
                row[j] = res[i - 1][j - 1] + res[i - 1][j];
            }
            res.push_back(row); // 把這一列加入結果中
        }
        return res;
    }
};

File: 0118-pascals-triangle/0118-pascals-triangle.py
class Solution:
    def generate(self, numRows: int) -> List[List[int]]:
        res = []
        for i in range(numRows):
            row = [1] * (i + 1) # 頭尾都是 1，中間預設先填 1
            for j in range(1, i): # 從第二個元素到倒數第二個
                row[j] = res[i - 1][j - 1] + res[i- 1][j]
            res.append(row)
        return res